traffic-sniffer/lib/request-metadata/sniffer-manager/mock-manager-controller.ts

```diff
@@ -0,0 +1,180 @@
+import { Express, NextFunction, Request, Response } from "express";
+import { SnifferManager } from "./sniffer-manager";
+import { MockNotFoundError } from "../sniffer/mock/exceptions";
+
+export class MockManagerController {
+  constructor(private readonly snifferManager: SnifferManager) {}
+
+  setup(app: Express) {
+    app.get(
+      "/sharkio/sniffer/action/getMocks",
+      async (req: Request, res: Response, next: NextFunction) => {
+        try {
+          const mocks = this.snifferManager.getAllMocks();
+          res.json(mocks).status(200);
+        } catch (e) {
+          console.error(e);
+          res.sendStatus(500);
+        }
+      }
+    );
+
+    app.get(
+      "/sharkio/sniffer/:port/mock",
+      async (req: Request, res: Response, next: NextFunction) => {
+        try {
+          const { port } = req.params;
+
+          const sniffer = this.snifferManager.getSniffer(+port);
+          if (sniffer !== undefined) {
+            const mocks = sniffer.getMockManager().getAllMocks();
+            res.send(mocks).status(200);
+          } else {
+            res.sendStatus(404);
+          }
+        } catch (e) {
+          console.error(e);
+          res.sendStatus(500);
+        }
```

Comment:
This part of code seems to very repeating so until we further refactor it, we can have it as a common method as such?

Reply:
We could check out express-promise-router